core/vault.go

```diff
@@ -0,0 +1,416 @@
+package core
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/starkandwayne/goutils/log"
+)
+
+type Vault struct {
+	URL            string
+	Token          string
+	EncryptionType string
+	Insecure       bool
+	HTTP           *http.Client
+}
+
+type VaultCreds struct {
+	SealKey        string `json:"seal_key"`
+	RootToken      string `json:"root_token"`
+	EncryptionType string `json:"encryption_type"`
+}
+
+var status struct {
+	Sealed bool `json:"sealed"`
+}
+
+func (vault *Vault) Init(store string) error {
+	initialized, err := vault.IsInitialized()
+	if err != nil {
+		return err
+	}
+
+	if initialized {
+		log.Infof("vault is already initialized")
+
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Print("Enter Master Password: ")
```

Comment:
The master password should be provided to the `Init()` function as a parameter.  Elsewhere, we need a new endpoint, `/v2/init` that takes the master password from the frontend (UI / CLI) and performs the initialization via `Ini()`